tests/fips/libtool/liboauth.pm

```diff
@@ -0,0 +1,102 @@
+# Copyright (C) 2021 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: liboauth test for FIPS integration test
+#
+# Description: liboauth is a collection of c functions implementing the http://oauth.net API.
+#              liboauth provides functions to escape and encode stings according to OAuth specifications
+#              and offers high-level functionality built on top to sign requests or verify signatures
+#              using either NSS or OpenSSL for calculating the hash/signatures.
+#
+# Maintainer: Ben Chou <bchou@suse.com>
+# Tags: poo#81260, tc#1767540
+
+use base "consoletest";
+use testapi;
+use utils;
+use utils "zypper_call";
+use strict;
+use warnings;
+use registration qw(add_suseconnect_product cleanup_registration register_product);
+
+sub run {
+    select_console 'root-console';
+
+    zypper_output();
+
+    # Since we use the functional group qcow2 which is regitered to proxyscc
+    # The Source RPM packages are all located in SCC source pool repository
+    # Step 1. SUSEConnect -d deregister from proxyscc
+    # Step 2. SUSEConnect -r <CODE> to SCC
+    # Step 3. SUSEConnect -p module to add DEV tool and Desktop App modules
+
+    # De-register the system from proxyscc due to source RPM package does not mirror to proxyscc
+    cleanup_registration();
+    zypper_output();
+
+    # Register to official SCC(https://scc.suse.com) to get source RPM packages
+    register_product();
+    zypper_output();
+
+    # Add Desktop_Applications_Module_15_SP1_x86_64
+    add_suseconnect_product("sle-module-desktop-applications");
+
```

Comment:
Why not use gnome image, then we don't have to add this product?

Reply:
Yes, I use the functional group qcow2 which is gnome image, but I de-register all of them from proxyscc, so I need to add them back here.